source/val/construct.h

```diff
@@ -91,6 +101,12 @@ class Construct {
     return type_ == ConstructType::kLoop || type_ == ConstructType::kSelection;
   }
 
+  using ConstructBlockSet = std::set<BasicBlock*, less_than_id>;
+
+  // Returns the basic blocks in this construct. This function should not
+  // be called before the exit block is set.
```

Comment:
Also requires dominators to have been computed.  That's because it relies on Function::GetBlockDepth